x-pack/plugins/siem/public/components/open_timeline/types.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/** The users who added a timeline to favorites */
+export interface FavoriteTimelineResult {
+  userId?: number | null;
+  userName?: string | null;
+  favoriteDate?: number | null;
+}
+
+export interface TimelineResultNote {
+  savedObjectId?: string | null;
+  note?: string | null;
+  updated?: number | null;
+  updatedBy?: string | null;
+}
+
+/** The results of the query run by the OpenTimeline component */
+export interface TimelineResult {
+  created?: number | null;
+  description?: string | null;
+  eventIdToNoteIds?: { [eventId: string]: string[] } | null;
```

Comment:
You can use an Advanced Type of Record here if you want to and even wrap it in a Readonly for two of these if you want to be more strict:

```ts
eventIdToNoteIds?: Readonly<Record<string, string[]>> | null;
pinnedEventIds?: Readonly<Record<string, boolean>> | null;
```

Ref:
https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#partial-readonly-record-and-pick
https://levelup.gitconnected.com/advanced-typescript-types-with-examples-1d144e4eda9e

Reply:
Great suggestion! I updated the type definitions throughout the project for both types.

- All of the updated type definitions use `Readonly`, except for the redux model
-`TimelineResult` (which represents the current wire format, and is subject to change) is the only variant that required `| null`.

Thanks!